include/vinecopulib/bicop/implementation/kernel.ipp

```diff
@@ -16,50 +17,80 @@ inline KernelBicop::KernelBicop()
   Eigen::VectorXd grid_points(m);
   for (size_t i = 0; i < m; ++i)
     grid_points(i) = -3.25 + i * (6.5 / static_cast<double>(m - 1));
-  grid_points = tools_stats::pnorm(grid_points);
-  
-  // move boundary points to 0/1, so we don't have to extrapolate
-  grid_points(0) = 0.0;
-  grid_points(m - 1) = 1.0;
-  
   interp_grid_ = std::make_shared<tools_interpolation::InterpolationGrid>(
-    grid_points,
+    tools_stats::pnorm(grid_points),
     Eigen::MatrixXd::Constant(m, m, 1.0) // independence
   );
 }
 
 inline Eigen::VectorXd
-KernelBicop::pdf_raw(const Eigen::Matrix<double, Eigen::Dynamic, 2>& u)
+KernelBicop::pdf_raw(const Eigen::MatrixXd& u)
+{
+  auto pdf = interp_grid_->interpolate(u);
+  auto trunc = [] (const double& p) {
+      return std::fmax(p, 1e-20);
```

Comment:
Why not `std::max` ?

Reply:
should be no difference, but it's getting replaced by `tools_eigen::trim` anyway